### ISSUE TITLE
Compiling Instructions for Raspberry Pi

### ISSUE THREAD

**Issue by Wade821** [Aloshi/EmulationStation#714]
`make -j 4 `

would make compiling much faster on the Raspberry Pi 3 as it uses all four cores rather than just 1 as noted on this [page - Getting Started](http://www.emulationstation.org/gettingstarted.html). I wouldn't call this an issue however, I just couldn't get a working email from this page [Feedback](http://www.emulationstation.org/index.html#feedback). 

That being said, I was working on making a script that would do everything needed even through restarts. I've got most of it done, I just ran into a slight issue I guess that I'll need to play around with more later. I'm in the process of just doing a run to see if it'll work this way, though it seems to be pretty high stress to the Pi.

FYI, I'm using the information I gathered from my own question on the scripts through a restart via this post [here.](https://serverfault.com/questions/879573/schedule-a-script-job-to-run-automatically-but-requires-user-interaction-input)

**Comment by iCrazyBlaze** [comment]
You could always do 
`make -j 2`
Like I did, which stops the Raspberry Pi from crashing, while installing ES faster.

**Comment by Wade821** [comment]
Well I've given up on the "faster" compile for now and just started to focus on why it isn't compiling. So here's a bunch of info on my progress thus far!

**Pi Model or other hardware:**  Raspberry Pi 3 (RPi3) Model B Quad-Core 1.2 GHz 1 GB RAM
**Power Supply used:** 5v 2.5A
**RetroPie Version Used:** N/A
**Built From:** Raspbian 4.9 Stretch with Desktop, 2017-11-29
**USB Devices connected:** 1 - iPazzPort Wireless Mini Keyboard with Touchpad for Android TV Box and Raspberry Pi 3 and HTPC KP-810-21S
**Controller used:** Not connected now as it doesn't compile, but XBOX 360 Wired controller
**Error messages received:** Multiple in LXTerminal, see logs listed below
**Guide used:** [EmulationStation: Installing on Raspberry Pi Stand-alone](https://emulationstation.org/gettingstarted.html#install_rpi_standalone)
**Attachment of config files:**	[Log 1 - Part 1](https://pastebin.com/5zzypVay) / [Log 1 - Part 2](https://pastebin.com/JkpGvpGx)
**How to replicate the problem:**	Using script, [part 01](https://pastebin.com/kJnL4HCb) and [part 02](https://pastebin.com/zeLFSFik) which is based off of the guide listed above. 

In the script, I commented out the *sudo reboot* line so I could observe all the output.  I don't know why it refuses to compile! Much frustration, I hope someone has some insight on how to do this. I've dug through the forums and read all sorts of things, with no success. I'm hoping someone can make better sense of it than I can. I've got a second log that's fully copied from start to end of the script part 02, however PasteBin is under heavy load and won't let me post it there for the time being.  So if it's needed, I made it as a [PDF](https://www.dropbox.com/s/rez7o1uvepfjvvl/Log%2002%20Full.pdf?dl=0) and placed it in dropbox.

Tried [this](https://retropie.org.uk/forum/topic/15661/emulationstation-fails-to-compile/3) to no avail. Installed *gcc-4.9 g++-4.9* in addition to the other dependencies listed in the guide. Changed line **cmake -DCMAKE_CXX_COMPILER=g++-4.7 .** to **cmake -DCMAKE_CXX_COMPILER=g++-4.9 .** instead, still doesn't work. 

I'm also trying to get help with compiling from the guide [here](https://retropie.org.uk/forum/topic/15970/emulation-station-for-raspbian-fails-to-complete-compile-at-100).

**Comment by dankcushions** [comment]
FYI retropie does not use this version of emulationstation, it uses this fork: https://github.com/retropie/EmulationStation

this version hasn't been updated for ~3 years. i don't know what version's source code you're using, but you're using a mixture of guides targeting different versions.

**Comment by Wade821** [comment]
I found out it was a fork the other night. I had forgotten up until then I could come here for further assistance, and was largely using the guide on [EmulationStation](https://emulationstation.org/gettingstarted.html#install_rpi_standalone). I do realize now that this is older and hasn't been updated, so I'm thinking of abandoning the script to install within Raspbian (since I've already got stuff installed there) and just downloading the Retropie image to setup Drawpile on that instead. I just wasn't sure if it was worth pursuing the guide on that page at the time.
